app/repositories/packages_repository.rb

```diff
@@ -1,162 +1,105 @@
+    payload[:isHidden] = payload.dig(:visibilityData, :isHidden)
+    payload.delete(:visibilityData)
+    content_type_enum = {
+      aggregatedfulltext: 1,
+      abstractandindex: 2,
+      ebook: 3,
+      ejournal: 4,
+      print: 5,
+      unknown: 6,
+      onlinereference: 7
+    }
+    if existing_package.is_custom
       payload[:contentType] = content_type_enum[payload[:contentType]&.downcase&.to_sym] || 6
+      payload[:packageName] = payload.delete(:name)
+    end
 
-      rmapi(:put, "/vendors/#{vendor_id}/packages/#{package_id}", json: payload)
+    request(:put, "/vendors/#{vendor_id}/packages/#{package_id}", json: payload)
 
-      status, body = rmapi(:get, "/vendors/#{vendor_id}/packages/#{package_id}")
-      Result.new(data: to_package(body), status: status)
-    end
+    status, body = request(:get, "/vendors/#{vendor_id}/packages/#{package_id}")
+    Result.new(data: to_package(body), status: status)
   end
 
   def destroy!(id)
-    request do
-      package = find!(id).data
-      package_validation = Validation::PackageDestroyParameters.new(package)
-      fail ValidationError, package_validation unless package_validation.valid?
-
-      vendor_id, package_id = id.split('-')
-      status, _no_content = rmapi(:put, "/vendors/#{vendor_id}/packages/#{package_id}", json: { isSelected: false })
+    vendor_id, package_id = id.split('-')
+    status, _no_content = request(:put, "/vendors/#{vendor_id}/packages/#{package_id}", json: { isSelected: false })
 
-      Result.new(data: {}, status: status)
-    end
+    Result.new(data: {}, status: status)
   end
 
   def where!(params)
-    request do
-      vendor_id = params.fetch(:vendor_id, nil)
-
-      if params[:filter]
-        fail BadRequest, 'Invalid filter parameter' unless params[:filter].respond_to?(:dig)
-        if params.dig(:filter, :custom)
-          # The 'custom' filter option is unique in that
-          # it is not passed through to RMAPI.
-          #
-          # TODO: remove mutation. params sent to rmapi should be completely
-          # mapped from scratch every time, not a munge of controller params
-          # (cowboyd 5/3/2018)
-          params[:filter].delete(:custom)
-
-          # All custom packages have the same vendor_id per tenant
-          vendor_id = Provider.configure(@config).provider_id
-        end
+    vendor_id = params.fetch(:vendor_id, nil)
+
+    # TODO: controller?
+    if params[:filter]
+      fail RequestError.new('Invalid filter parameter', 400) unless params[:filter].respond_to?(:dig)
+      if params.dig(:filter, :custom)
+        # The 'custom' filter option is unique in that
+        # it is not passed through to RMAPI.
+        #
+        # TODO: remove mutation. params sent to rmapi should be completely
+        # mapped from scratch every time, not a munge of controller params
+        # (cowboyd 5/3/2018)
+        params[:filter].delete(:custom)
+
+        # All custom packages have the same vendor_id per
```

Comment:
per what? Seems like it could be expanded upon a little.

Reply:
looks like i snipped a line here by mistake ✂️ thanks for catching that